src/ui/src/common/extension_command.ts

```diff
@@ -41,53 +61,52 @@ export declare interface AdapterConvertCommand extends ExtensionCommand {
 }
 
 /** Adapter's "convert" command response. */
-export declare interface AdapterConvertResponse extends ExtensionResponse {
-  graphs?: Graph[];
-  graphCollections?: GraphCollection[];
-}
+export type AdapterConvertResponse = ExtensionResponse;
 
 /** Adapter's "override" command. */
 export declare interface AdapterOverrideCommand extends ExtensionCommand {
   cmdId: 'override';
-  modelPath: string;
   settings: {
     graphs: Graph[];
     changes: ChangesPerNode;
   };
-  deleteAfterConversion: boolean;
 }
 
 /** Adapter's "override" command response. */
-export declare interface AdapterOverrideResponse extends ExtensionResponse {
+export type AdapterOverrideResponse = ExtensionResponse<[{
   success: boolean;
-  graphs?: Graph[];
-}
+}], never>;
 
 /** Adapter's "execute" command. */
 export declare interface AdapterExecuteCommand extends ExtensionCommand {
   cmdId: 'execute';
-  modelPath: string;
-  settings: Record<string, any>;
-  deleteAfterConversion: boolean;
 }
 
-/** Adapter's "execute" command response. */
-export declare interface AdapterExecuteResponse extends ExtensionResponse, ExecutionCommand {
+/** Adapter's "execute" results inside the response. */
+export interface AdapterExecuteResults {
+  stdout: string;
+  log_file: string;
+  perf_trace?: string;
+  perf_data?: NodeDataProviderData
```

Comment:
I think that these should be part of the convert command. As i understood the diagram you showed on last weeks sync:
Execute is called -> sends back only status if execution start is successful.
Status check is called (multiple times, until isDone is true) -> sends back progress info and logs
Once isDone == true is received, another convert call is made -> sends back the "new / optimized" graph with perf data.

Am i missing something?

Reply:
It is correct.
I'll remove that property

Reply:
Yup, I think we decided that's how we will process the data. I'll wait on @odjuricicTT for his PR with that functionality to merge in the perf data support.